src/urdb.cpp

```diff
@@ -457,372 +528,522 @@ bool OpenEI::RetrieveUtilityRateData(const wxString &guid, RateData &rate, wxStr
-	wxJSONValue ers_periods = val.Item("energyratestructure");
-	if (ers_periods.IsArray())
-	{
-		for (int period = 0; period < ers_periods.Size(); period++)
-		{
-			wxJSONValue ers_tier = ers_periods[period];
-			if (ers_tier.IsArray())
-			{
-				for (int tier = 0; tier < ers_tier.Size(); tier++)
-				{
-					num_ec_rows++;
+	if (val.HasMember("energyratestructure")) {
+		int num_ec_rows = 0; // default to one for each month
+		//	wxJSONValue ers_periods = val.Item("energyratestructure");
+		auto& ers_periods = val["energyratestructure"];
+		if (ers_periods.IsArray())	{
+			for (size_t period = 0; period < ers_periods.Size(); period++)	{
+				auto& ers_tier = ers_periods[period];
+				if (ers_tier.IsArray())	{
+					for (size_t tier = 0; tier < ers_tier.Size(); tier++)	{
+						num_ec_rows++;
+					}
 				}
 			}
 		}
-	}
 
-	if (num_ec_rows > 0)
-	{
-		rate.EnergyStructure.resize_fill(num_ec_rows, 6, 0.0);
-
-	// next, assign rate values
-		int es_row = 0;
-		for (int period = 0; period < ers_periods.Size(); period++)
+		if (num_ec_rows > 0)
 		{
-			wxJSONValue ers_tier = ers_periods[period];
-			for (int tier = 0; tier < ers_tier.Size(); tier++)
-			{
-				double max = json_double(ers_tier[tier].Item("max"), 1e38, &rate.HasEnergyCharge);
-				double buy = json_double(ers_tier[tier].Item("rate"), 0.0, &rate.HasEnergyCharge);
-				double sell = json_double(ers_tier[tier].Item("sell"), 0.0, &rate.HasEnergyCharge);
-				double adj = json_double(ers_tier[tier].Item("adj"), 0.0, &rate.HasEnergyCharge);
-				wxString units = json_string(ers_tier[tier].Item("unit"));
-                rate.EnergyUnits = "";
-				int iunits = 0; // assume kWh if units not provided
-				if (units.Lower() == "kwh")
-					iunits = 0;
-				else if (units.Lower() == "kwh/kw")
-					iunits = 1;
-				else if (units.Lower() == "kwh daily")
-					iunits = 2;
-				else if (units.Lower() == "kwh/kw daily")
-					iunits = 3;
-                else {
-                    rate.Unused.HasUnusedItems = true;
-                    rate.EnergyUnits = units;
-                }
-				rate.EnergyStructure.at(es_row, 0) = period + 1;
-				rate.EnergyStructure.at(es_row, 1) = tier + 1;
-				rate.EnergyStructure.at(es_row, 2) = max;
-				rate.EnergyStructure.at(es_row, 3) = iunits;
-				rate.EnergyStructure.at(es_row, 4) = buy+adj;
-				rate.EnergyStructure.at(es_row, 5) = sell;
-				es_row++;
+			rate.EnergyStructure.resize_fill(num_ec_rows, 6, 0.0);
+
+			// next, assign rate values
+			int es_row = 0;
+			for (size_t period = 0; period < ers_periods.Size(); period++)	{
+				auto& ers_tier = ers_periods[period];
+				for (size_t tier = 0; tier < ers_tier.Size(); tier++) {
+//					double max = json_double(ers_tier[tier].Item("max"), 1e38, &rate.HasEnergyCharge);
+					double max = 1e38;
+					if (ers_tier[tier].HasMember("max"))
+						if (ers_tier[tier]["max"].IsDouble())
+							max = ers_tier[tier]["max"].GetDouble();
+					double buy = 0.0;
+					if (ers_tier[tier].HasMember("buy"))
```

Comment:
I think this should be "rate" instead of "buy" see error in main comment:

![image](https://github.com/user-attachments/assets/7613f59c-4254-44db-8961-1452ea4ffd0f)